sdk/src/main/java/com/bugsnag/android/ErrorReader.java

```diff
@@ -0,0 +1,439 @@
+package com.bugsnag.android;
+
+import android.support.annotation.NonNull;
+import android.util.JsonReader;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+class ErrorReader {
+
+    /**
+     * Parses an {@link Error} cached as JSON into an Error object.
+     *
+     * @throws IOException if the file cannot be parsed into a valid JSON object,
+     *                     such as if the JSON syntax is invalid or a required
+     *                     field is missing.
+     */
+    static Error readError(@NonNull Configuration config, @NonNull File errorFile)
```

Comment:
The benefit of it being static is that there isn't any state (shared or otherwise) to be mocked. Its obvious what it does and that all related logic/state is within the method itself.
Adding additional state is something we want to avoid as long as possible, so there's no need to make it easier before we actually need it.